integration-test/src/test/java/org/airsonic/test/Scanner.java

```diff
@@ -0,0 +1,137 @@
+
+        Long waitTime = 30000L;
+        Long sleepTime = 1000L;
+        while (waitTime > 0 && isScanning()) {
+            waitTime -= sleepTime;
+            Thread.sleep(sleepTime);
+        }
+
+        Assert.assertFalse(isScanning());
+    }
+
+    private static boolean isScanning() {
+        return rest.getForObject(
+                addRestParameters(UriComponentsBuilder.fromHttpUrl(SERVER + "/rest/getScanStatus"))
+                        .queryParam("f", "json").toUriString(),
+                SubsonicResponse.class)
+            .getScanStatus().isScanning();
+    }
+
+    public static void uploadToDefaultMusicFolder(Path directoryPath, String relativePath) {
+        Path dest = Paths.get(DEFAULT_MUSIC, relativePath);
+        try {
+            FileUtils.copyDirectory(directoryPath.toFile(), dest.toFile(), false);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static List<Child> getMediaFilesInMusicFolder() {
+        List<MusicFolder> musicFolder = rest.getForObject(
+                addRestParameters(UriComponentsBuilder.fromHttpUrl(SERVER + "/rest/getMusicFolders"))
+                        .queryParam("f", "json")
+                        .toUriString(),
+                SubsonicResponse.class)
+            .getMusicFolders().getMusicFolder();
+
+        MusicFolder music = musicFolder.stream().filter(folder -> Objects.equals(folder.getName(), "Music")).findFirst()
+                .orElseThrow(() -> new RuntimeException("No top level folder named Music"));
+        return getMediaFiles(music.getId());
+    }
+
+    private static List<Child> getMediaFiles(int folderId) {
+        return rest.getForObject(
+                addRestParameters(UriComponentsBuilder.fromHttpUrl(SERVER + "/rest/getIndexes"))
+                        .queryParam("f", "json")
+                        .queryParam("musicFolderId", folderId)
+                        .toUriString(),
+                SubsonicResponse.class)
+            .getIndexes().getChild();
+    }
+
+    public static byte[] getMediaFileData(String mediaFileId) {
+        HttpHeaders headers = new HttpHeaders();
+        headers.setAccept(MediaType.parseMediaTypes("audio/webm,audio/ogg,audio/wav,audio/*;"));
+        ResponseEntity<byte[]> response = rest.exchange(
+                addRestParameters(UriComponentsBuilder.fromHttpUrl(SERVER + "/rest/stream"))
+                        .queryParam("id", mediaFileId)
+                        .toUriString(),
+                HttpMethod.GET,
+                new HttpEntity<>(headers),
+                byte[].class);
+
+        assertThat(response.getBody()).hasSize((int) response.getHeaders().getContentLength());
+        return response.getBody();
+    }
+
+    @JsonRootName(value = "subsonic-response")
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    @JsonIgnoreProperties(ignoreUnknown = true)
+    public static class SubsonicResponse extends Response {
+    }
```

Comment:
This wrapper added because I just wanted to make do with Jackson instead of adding additional JAXB dependencies to deal with the incoming xml

Reply:
Can't we use the existing classes?

Reply:
It does use the existing class (*extends* Response). You also need a _binding_ library that binds xml to Java classes. You may use cxf. I thought it easier (since it was already in the classpath) to just use Jackson. But you need some annotations to account for some xsd elements, especially since the code generation doesn't seem to create self-sufficient classes (like mark an _XmlRootElement_ on Response). If you want, you can use a JAXB binder other than Jackson